### PR TITLE
Volunteer page milestone 4: Address some designer feedbacks and implement front-end email logics.

### PR DESCRIPTION
**Pull request by hoangviet1993**
- [x] Page circles jumping around and blocking quote text box
- [x] Inconsistent form's text font (oversized "Message")
- [x] Front-end logics for email functionality.
- [ ] Fixed size quote box: I decided to not do this. Will discuss this over triage meeting on Saturday(09/01). 
Some pics and a gif!

Close up view of form:
screen shot 2018-09-01 at 12 01 29 am

Close up view of profile carousel:
screen shot 2018-09-01 at 12 02 41 am

Gif:

Can provide full screenshots of desktop and mobile on request...but it is largely the same with some minor visual improvement shown above.

### CODE REVIEW

**Comment by seanlip** [comment]
Hm, only one UI question. Would it be possible to position the carousel arrows a bit above the bottom margin of the picture? Sometimes they block the text.

Otherwise I think everything looks great!

File: static/assets/constants.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/* eslint quote-props: [2, "always"]*/
+/* Don't modify anything outside the {} brackets.
+ * Insides of the {} brackets should be formatted as a JSON object.
+ * JSON rules:
+ * 1. All keys and string values must be enclosed in double quotes.
+ * 2. Each key/value pair should be on a new line.
+ * 3. All values and keys must be constant, you can't use any Javascript
+ *    functions.
+ */
+var constants = {
```
Comment (seanlip) [inline static/assets/constants.js]: Actually I think you can just declare the constants on oppiaFoundationWebsite directly?

In general assets/constants.js is only meant to be for constants that are shared between frontend and backend. Here, though, the backend doesn't use these.

File: static/pages/volunteer/Volunteer.js [hunk 2]
```diff
@@ -12,8 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-oppiaFoundationWebsite.controller('VolunteerPage', ['$scope', '$document',
-  '$window', function($scope, $document, $window) {
+oppiaFoundationWebsite.controller('VolunteerPage', [
+  '$scope', '$document', '$http', '$mdDialog', '$log', '$window',
+  'MAILHANDLER_URL', 'VOLUNTEER_EMAIL_TYPE', 'THANKYOU_MESSAGE',
```
Comment (seanlip) [inline static/pages/volunteer/Volunteer.js]: Line breaks below should match those here, for easy verification that the dependency lists match.

File: static/pages/volunteer/Volunteer.js [hunk 3]
```diff
@@ -59,4 +63,25 @@ oppiaFoundationWebsite.controller('VolunteerPage', ['$scope', '$document',
         throw Error('No such element');
       }
     };
+    $scope.submitContactUsForm = function(
+        fullName, email, comment, event) {
```
Comment (seanlip) [inline static/pages/volunteer/Volunteer.js]: 'event' is a keyword in some old browsers, let's change it to evt.

File: static/assets/constants.js (hunk 1 above)
Comment (hoangviet1993) [reply static/assets/constants.js]: I thought constants.js was a neat idea, kind of a parity between config/feconf for backend and constants.js for frontend. I guess I will do it the normal way then.

Comment (hoangviet1993) [reply static/assets/constants.js]: Done!

File: static/pages/volunteer/Volunteer.js (hunk 2 above)
Comment (hoangviet1993) [reply static/pages/volunteer/Volunteer.js]: Done!

File: static/pages/volunteer/Volunteer.js (hunk 3 above)
Comment (hoangviet1993) [reply static/pages/volunteer/Volunteer.js]: Done!

**Comment by hoangviet1993** [comment]
So the "carousel control layer"(the left,right arrow, the page circles) is on top of the "carousel content layer"(the photos, the text) so I am not sure as to how it would make the left/right arrow match the bottom of the profile image as they are treated as one single canvas.
(I can only make them move positionally based on percentage)
As a compromise, the control layer has a fixed size now (650px) and the carousel buttons are 20% top of that 650px.
Close-up photo of carousel now:
screen shot 2018-09-01 at 9 36 56 am
Carousel in action!


PTAL @seanlip

**Comment by seanlip** [comment]
Confirming carousel looks good to me, thanks!

**Pull request merged**